src/ImageSharp/Formats/Jpg/Components/Bytes.cs

```diff
@@ -3,41 +3,168 @@
 // Licensed under the Apache License, Version 2.0.
 // </copyright>
 
+using System;
+using System.Buffers;
+using System.IO;
+using System.Runtime.CompilerServices;
+
 namespace ImageSharp.Formats
 {
     /// <summary>
     /// Bytes is a byte buffer, similar to a stream, except that it
     /// has to be able to unread more than 1 byte, due to byte stuffing.
     /// Byte stuffing is specified in section F.1.2.3.
     /// </summary>
-    internal class Bytes
+    internal struct Bytes : IDisposable
     {
         /// <summary>
         /// Initializes a new instance of the <see cref="Bytes"/> class.
         /// </summary>
-        public Bytes()
+        //public Bytes()
+        //{
+        //    this.Buffer = new byte[4096];
+        //    this.I = 0;
+        //    this.J = 0;
+        //    this.UnreadableBytes = 0;
+        //}
+
+        private static readonly ArrayPool<byte> ArrayPool = ArrayPool<byte>.Create(4096, 50);
+
+        public static Bytes Create()
         {
-            this.Buffer = new byte[4096];
-            this.I = 0;
-            this.J = 0;
-            this.UnreadableBytes = 0;
+            return new Bytes
+            {
+                Buffer = ArrayPool.Rent(4096)
+            };
         }
 
         /// <summary>
         /// Gets or sets the buffer.
         /// buffer[i:j] are the buffered bytes read from the underlying
         /// stream that haven't yet been passed further on.
         /// </summary>
-        public byte[] Buffer { get; set; }
+        public byte[] Buffer;
 
-        public int I { get; set; }
+        public int I;
```

Comment:
I *think* the values represent the last two used index positions within the block but I'd have to have another read through. 

No worries, we'll figure it out eventually.